### DISCUSSION TITLE
Function Returning Tiles

### DISCUSSION THREAD

**Discussion by aemonm** [developmentseed/tipg#101] [Q&A] [answered]
Following [the docs](https://developmentseed.org/tipg/advanced/functions/), I created a function to return tiles like so:
```
CREATE FUNCTION get_slicks_by_aoi(IN z int,
    IN x int,
    IN y int,
    IN aoi_id int,
    OUT slick_id bigint,
    OUT geom geography
) RETURNS SETOF RECORD AS $$
    SELECT sp.id slick_id, sp.geometry geom FROM public.slick_plus sp join slick_to_aoi sta on sta.aoi = aoi_id where ST_Intersects(sp.geometry, ST_Transform(ST_TileEnvelope(z, x, y), 4326));
$$ LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
```

When I try and run this function, like : `GET /collections/public.get_slicks_by_aoi/tiles/4/4/3?aoi_id=221` I'm get an error: `invalid input for query argument $10: 4 (expected str, got int)` .

I might be doing something wrong, but this could be also a bug. Any ideas?
N.B. We are using a [commit pinned](https://github.com/developmentseed/tipg/commit/a435d30db9aefd8094ece134681e9c562a6f696a) version of TiPG:

**Comment by vincentsarago** [comment, marked as answer]
thanks for starting the discussion @aemonm,

It might be pretty hard for us to debug bug as far as I know we did a lot of changed in the `Functional` Layer code base and fix some issue since the commit you shared. If possible it would be nice if you could test with `0.2.0` or `0.3.0`

**Reply by aemonm** [reply]
This issue seems to be resolved after upgrading to 0.3.

**Reply by vincentsarago** [reply]
🙏
